jvm/src/main/java/com/microsoft/bond/protocol/TaggedProtocolReader.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+
+import java.math.BigInteger;
+
+/**
+ * Reads a serialized payload.
+ */
+public interface TaggedProtocolReader {
+    class ReadFieldResult {
```

Comment:
If I think really hard about it, I can come up with a use case for non-finality. We can always definalicize them later, though. Changed.